gir/pkgconfig/pkgconfig.go

```diff
@@ -1,105 +1,152 @@
-	if err := ensure(); err != nil {
-		return nil, err
+// Variable returns a map of map[pkg]value containing the given
+// variable value for each requested package.  It is not an error for
+// a variable to be unset or empty; ret[pkg] is an empty string if
+// that package does not have the variable set.
+func Variable(varname string, pkgs ...string) (map[string]string, error) {
+	if len(pkgs) == 0 {
+		return nil, nil
 	}
 
-	args := []string{"--variable=includedir"}
-	args = append(args, pkgs...)
-
-	out, err := exec.Command(path, args...).Output()
-	if err != nil {
+	cmdline := append([]string{"pkg-config",
+		// Don't be opaque when we fail.
+		"--print-errors",
+		// On Nix, there are "normal" and "dev" split
+		// packages, with different prefixes.  The "prefix"
+		// variable is set to the normal package's prefix,
+		// while the .pc, .h and .gir files are installed
+		// relative to the dev package's prefix.  So tell
+		// pkg-config to override the prefix variable with the
+		// location of the .pc file.  An alternative way of
+		// getting the dev package's suffix would be to
+		// inspect the "includedirs" variable and trim off the
+		// "/include(/.*)?$" suffix.
+		"--define-prefix",
+		// main
+		"--variable=" + varname, "--"}, pkgs...)
+	var stdout strings.Builder
+	cmd := exec.Command(cmdline[0], cmdline[1:]...)
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
 		var exitErr *exec.ExitError
 		if !errors.As(err, &exitErr) {
 			return nil, fmt.Errorf("pkg-config failed: %w", err)
 		}
-
-		return nil, fmt.Errorf(
-			"pkg-config failed with status %d:\n%s",
-			exitErr.ExitCode(), exitErr.Stderr,
-		)
+		return nil, fmt.Errorf("pkg-config failed with status %d:\n%s",
+			exitErr.ExitCode(), exitErr.Stderr)
 	}
 
-	return strings.Fields(string(out)), nil
-}
+	ret := make(map[string]string, len(pkgs))
+	stdoutStr := strings.TrimRight(stdout.String(), "\n")
+	if stdoutStr == "" {
+		for i := range pkgs {
+			ret[pkgs[i]] = ""
+		}
+	} else {
+		vals := strings.Split(stdoutStr, " ")
+		if len(vals) != len(pkgs) {
+			return nil, fmt.Errorf("%v returned %d values, but expected %d",
+				cmdline, len(vals), len(pkgs))
+		}
 
-func findInclude(dir string) string {
-	parts := strings.Split(dir, string(filepath.Separator))
-	for i := len(parts) - 1; i >= 0; i-- {
-		if parts[i] == "include" {
-			return "/" + filepath.Join(parts[:i]...)
+		for i := range pkgs {
+			ret[pkgs[i]] = vals[i]
 		}
 	}
-	return ""
+
+	return ret, nil
 }
 
-// FindGIRFiles finds gir files from the given list of pkgs.
-func FindGIRFiles(pkgs ...string) ([]string, error) {
-	includeDirs, err := IncludeDirs(pkgs...)
+// VariableOrElse is like Variable, but if a package has an
+// empty/unset value, then that empty value is replaced with the value
+// that is returned from the fn function.
+func VariableOrElse(varname string, fn func(...string) (map[string]string, error), pkgs ...string) (map[string]string, error) {
```

Comment:
The chaining is cool, but `func(...string) (map[string]string, error)` should be its own type throughout this package. Perhaps a `PathMapGenerator` type?

Reply:
I think I'm going with `ValueSource`.

Reply:
That should at least have a `Func` suffix to the type name. Why `ValueSource`? What "value"?

Reply:
Done; I used the name `ValueSource`, which hopefully is clear given that it mirrors the new `Values()` name of the `Variable()` function.